core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -117,6 +117,12 @@ public function __construct($strTable, $arrModule=array())
 
 		$this->intId = Input::get('id');
 
+		if (!is_numeric($this->intId)) {
+			$this->intId = null;
```

Comment:
The type annotation for `DataContainer#$intId` currently says `string|int`, sure this should be `null`?

Can't we just always cast to `int` in `DC_Table`?

Reply:
It can be string within `DC_Folder` for example.

Reply:
Input::get() returns null, if the parameter is not set, so I thought null would be reasonable.

Reply:
> It can be string within `DC_Folder` for example.

This change would be only for `DC_Table`